services/history/src/middleware/jwt.ts

```diff
@@ -0,0 +1,26 @@
+import jwt from 'jsonwebtoken';
+import { NextFunction, Request, Response } from 'express';
+import { handleUnauthorized } from '../utils/responses';
+import config from '../config';
+import { userSchema } from '../types/request';
+
+export async function verifyAccessToken(req: Request, res: Response, next: NextFunction) {
+    const authHeader = req.headers['authorization'];
+    if (!authHeader) {
+        handleUnauthorized(res, 'Authenticated failed');
+        return;
+    }
+
+    const token = authHeader.split(' ')[1];
+
+    jwt.verify(token, config.JWT_SECRET, async (err, user) => {
+        const result = userSchema.safeParse(user);
+        if (err || result.error) {
+            handleUnauthorized(res, 'Authentication failed');
+            return;
+        }
+
+        req.user = result.data;
+        next();
+    });
```

Comment:
yep so `(err, user)` would probably be written better as `(err, payload)`, where `payload` is the decoded jwt token payload and `err` would be present if there were any issues parsing the jwt token such as invalid format or past the expiry time. Frankly I'm not too sure if `userSchema` is the full schema of what's in the `payload` object, but the nice thing is zod does parsing (validate + filter to keep only the fields you specify) instead of just validation.

Here's how the `accessToken` is generated in the user service:

```typescript
const accessToken = jwt.sign(
            {
                id: user.id,
                username: user.username,
                role,
            },
            config.JWT_SECRET,
            {
                expiresIn: '1d',
            },
        );
```